src/main/java/fr/greencodeinitiative/php/checks/AvoidTryCatchWithFileOpenedCheck.java

```diff
@@ -0,0 +1,158 @@
+                    break;
+                case DO_WHILE_STATEMENT:
+                    visitStatementsList(Arrays.asList(((DoWhileStatementTree) stmt).statement()));
+                    break;
+                case FOREACH_STATEMENT:
+                case ALTERNATIVE_FOREACH_STATEMENT:
+                    visitStatementsList(((ForEachStatementTree) stmt).statements());
+                    break;
+                case CASE_CLAUSE:
+                    visitStatementsList(((CaseClauseTree) stmt).statements());
+                    break;
+                case SWITCH_STATEMENT:
+                    visitSwitchStatement((SwitchStatementTree) stmt);
+                    break;
+                case DEFAULT_CLAUSE:
+                    visitStatementsList(((DefaultClauseTree) stmt).statements());
+                    break;
+                case TRY_STATEMENT:
+                    visitTryStatement((TryStatementTree) stmt);
+                    break;
+                default:
+            }
+        }
+    }
+
+    private void visitIfStatement(IfStatementTree ifStatement) {
+        visitStatementsList(ifStatement.statements());
+
+        if (ifStatement.elseClause() != null) {
+            visitStatementsList(ifStatement.elseClause().statements());
+        }
+
+        if (ifStatement.elseifClauses() != null) {
+            for (ElseifClauseTree stmtElseIf : ifStatement.elseifClauses()) {
+                visitStatementsList(stmtElseIf.statements());
+            }
+        }
+    }
+
+    private void visitTryStatement(TryStatementTree tryStatement) {
+        for (CatchBlockTree stmtCatch : tryStatement.catchBlocks()) {
+            visitStatementsList(stmtCatch.block().statements());
+        }
+
+        if (tryStatement.finallyBlock() != null) {
+            visitStatementsList(tryStatement.finallyBlock().statements());
+        }
+    }
+
+    private void visitSwitchStatement(SwitchStatementTree switchStatement) {
+        for (SwitchCaseClauseTree switchCaseStmt : switchStatement.cases()) {
+            visitStatementsList(switchCaseStmt.statements());
+        }
+    }
+
+    private void visitExpressionStatement(ExpressionTree exprTree){
+        if (exprTree.is(Tree.Kind.FUNCTION_CALL)) { // si directement "function call"
+            visitCallExpression((FunctionCallTree) exprTree);
+        } else if (exprTree.is(Tree.Kind.ASSIGNMENT)) { // ou si assignment
+            AssignmentExpressionTree assignTree = (AssignmentExpressionTree) exprTree;
+            if (assignTree.value().is(Tree.Kind.FUNCTION_CALL)) { // et si "function call"
+                visitCallExpression((FunctionCallTree) assignTree.value());
+            }
+        }
+    }
+
+    private void visitCallExpression(FunctionCallTree functionCall){
+        String funcName = getFunctionNameFromCallExpression(functionCall);
+        if (funcName.startsWith("PDF_open")
+                || "fopen".equals(funcName)
+                || "readfile".equals(funcName)) {
```

Comment:
Methods are not case-sensitive in PHP, so maybe we could use `equalsIgnoreCase` instead?
The idea is to handle more cases if some developers use uppercase

Reply:
please check https://github.com/green-code-initiative/ecoCode-php/pull/10/commits/ed98eb7a9d661105a4301c09ebff951b390af073